### DISCUSSION TITLE
Possible that a user sees only "his" data?

### DISCUSSION THREAD

**Discussion by mattbue** [netbox-community/netbox#17538] [General]
Hi everyone, 
I have multiple customers from which I manage their data within one Netbox installation.

Now I'm thinking about giving these customers access to my netbox.

But is it possible that the customer specific user only sees the data which is assigned to him?

I already tried some configurations but I could get it that e.g. one customer sees only his VLANs and not the VLANs from another customer.

Thanks

**Comment by candlerb** [comment]
Is this a read-only view?

Are you using "tenant" to identify objects that belong to a single customer (for example, Devices?) In that case it may be possible for you to create permissions rules to view objects with a particular tenant.

But this breaks down for things which are shared between tenants, e.g. racks. If you want a customer to see all racks where they have equipment installed, but not other racks, then I think the best you can do is apply tags (one tag per customer) and permission rules which allow access only to items with a given tag. But it's then up to you manually to add and remove tags as necessary.

In any case, you have to create a new set of permission rules for each customer, since the permission rules hard-code one particular tenant or tag. It's not pretty.

Depending on what info they want, it might be better to set up some nightly exports per customer.

**Reply by DanSheps** [reply]
On some of the FK associations (Racks) you might be able to use the reverse_query field to get if the rack is viewable by a tenant (for shared racks).

It can get messy but I think it is 100% do-able.

**Comment by Myke** [comment]
I too would love for my tenants to be able to see assets that are allocated to them.
